cmd/freezeproxy/main.go

```diff
@@ -25,7 +25,7 @@ func main() {
 	log.Println("Connect to freeze daemon on:", hostIP)
 
 	var tokenCfg Token
-	go wait.PollInfinite(time.Minute, func() (done bool, err error) {
+	go wait.PollImmediateInfinite(time.Minute, func() (done bool, err error) {
```

Comment:
now that Im looking at this, Im wondering what the point of the PollImmediate (and now PollImmediateInfinite) is since we do a log.Fatal on any errors (which will crash the container). Wonder if we should just make this a `for {}` loop instead here?

Reply:
understood, but couldn't we achieve that by changing `wait.PollInfinite` in to a regular `for {}` loop rather than `PollImmediateInfinite`? i.e. something like (untested):

~~~~
go func() {
  for {
        token, err := ioutil.ReadFile("/var/run/projected/token")
 	if err != nil {
 		log.Fatal("could not read token", err)
	}
	tokenCfg.Set(string(token))
	log.Println("refresh token...")
  }
}()
~~~~

would be a little simpler and clearer

Reply:
there's actually a small bug here still because with all of these approaches there's no guarantee the go routine will run the first iteration of the loop before handling a user request (since the goroutine happens in parallel to everything else). This PR makes the window much smaller, but we probably need to do an initial read of the token outside the goroutine to complete solve the bug.

Reply:
Thanks for the guidance.
How about adding a wait for token is not empty before first send pause?
i.e.
```golang
	wait.PollImmediateInfinite(time.Second, func() (done bool, err error) {
		if tokenCfg.Get() != "" {
			return true, nil
		}
		return false, nil
	})
	// start paused.
	pause()
```

Or is there any other recommended way？

Reply:
I think Id just pull the refresh in to a `refreshToken` method, and then we could just do `refreshToken(); go func() { for { time.Sleep(1 * time.Minute); refreshToken() } }()`

Reply:
(otherwise you may have to wait up to a second for the container to start due to the polling)

Reply:
I have a question, can we confirm that `refreshToken` is executed after waiting for a second?
Would it be better to use `PollImmediateInfinite` to confirm the completion of `refreshToken` execution?

Reply:
why would you need to confirm anything if you execute it once synchronously, before the goroutine? If you wait for a second to confirm it's ready, the container might take an extra second to start up, which would be unfortunate.

Reply:
@julz Thanks for your review! the PR has been updated